src/lua/protobuf.lua

```diff
@@ -545,6 +581,39 @@ validate_scalar = function(field_def, value)
     end
 end
 
+local function validate_decoded_type(protocol, field_def, wire_type)
+    local protobuf_wire_types = {[0] = 'varint', [1] = 'I64',
+        [2] = 'len', [5] = 'I32'}
+    if protobuf_wire_types[wire_type] == nil then
+        error(('Decoding data contains incorrect protobuf type: %d'):format(
+            wire_type))
+    end
+    if field_def == nil then
+        return
+    end
+    local exp_type = field_def.type
+    local exp_wire_type = WIRE_TYPE_VARINT
+    if exp_type == 'string' or exp_type == 'bytes' then
+            exp_wire_type = WIRE_TYPE_LEN
+    elseif exp_type == 'double' or exp_type == 'fixed64' or
+        exp_type == 'sfixed64' then
+            exp_wire_type = WIRE_TYPE_I64
+    elseif exp_type == 'float' or exp_type == 'fixed32' or
+        exp_type == 'sfixed32' then
+            exp_wire_type = WIRE_TYPE_I32
```

Comment:
Can we include this mapping into `scalars`?

So, we can do something like the following:

```lua
local scalar_def = scalars[field_def.type]
scalar_def.wire_type -- this is the expected wire type
```